cmd/tailscale/netcheck.go

```diff
@@ -51,15 +51,24 @@ func runNetcheck(ctx context.Context, args []string) error {
 	}
 	fmt.Printf("\t* MappingVariesByDestIP: %v\n", report.MappingVariesByDestIP)
 	fmt.Printf("\t* HairPinning: %v\n", report.HairPinning)
-	fmt.Printf("\t* Nearest DERP: %v (%v)\n", report.PreferredDERP, c.DERP.LocationOfID(report.PreferredDERP))
-	fmt.Printf("\t* DERP latency:\n")
-	var ss []string
-	for s := range report.DERPLatency {
-		ss = append(ss, s)
-	}
-	sort.Strings(ss)
-	for _, s := range ss {
-		fmt.Printf("\t\t- %s = %v\n", s, report.DERPLatency[s])
+
+	// When DERP latency checking failed,
+	// magicsock will try to pick the DERP server that
+	// most of your other nodes are also using
+	if len(report.DERPLatency) == 0 {
+		fmt.Printf("\t* DERP latency checking failed, magicsock will pick the DERP server based on popularity\n")
```

Comment:
Hmm, we should probably try to match the writing style of the other output of netcheck. How about:

`\t* Nearest DERP: unknown (no response to latency probes)`

It doesn't explain that DERP might still work, but I think that's okay. This is debugging output, and I think @bradfitz was going to add an explicit test that connects to DERP over TCP anyway.

Reply:
Thank you!